sdk/synapse/test-resources-post.ps1

```diff
@@ -0,0 +1,26 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+# This script is used to generate the Test Configuration file for Storage live tests.
+# It is invoked by the https://github.com/Azure/azure-sdk-for-net/blob/main/eng/New-TestResources.ps1
+# script after the ARM template, defined in https://github.com/Azure/azure-sdk-for-net/blob/arm-template-storage/sdk/storage/test-resources.json, 
+# is finished being deployed. The ARM template is responsible for creating the Storage accounts needed for live tests.
+
+param (
+    [hashtable] $DeploymentOutputs,
+    [string] $TestApplicationOid,
+    [switch] $CI
+)
+
+if ($CI) { exit }
+
+Import-Module Az.Synapse
+
+$roleAssignment = (Get-AzSynapseRoleAssignment -WorkspaceName $DeploymentOutputs.AZURE_SYNAPSE_WORKSPACE_NAME -ObjectId $TestApplicationOid)
+
+if (!$roleAssignment) {
+    New-AzSynapseRoleAssignment `
+        -WorkspaceName $DeploymentOutputs.AZURE_SYNAPSE_WORKSPACE_NAME `
+        -RoleDefinitionName "Synapse Administrator" `
```

Comment:
Is there any risk of the name changing? If so we could use the RoleDefinitionId instead.

Reply:
It's documented here: https://docs.microsoft.com/en-us/azure/synapse-analytics/security/synapse-workspace-synapse-rbac-roles ... I don't see RoleDefinitionIds there - do we know those would be more stable?

Reply:
I don't know if they are more stable, but I've seen role names change prior to GA in the past. The role definition Id would tend not to change. Probably not worth worrying too much about it.